zproject/dev_settings.py

```diff
@@ -89,6 +89,38 @@
 # Set this True to send all hotspots in development
 ALWAYS_SEND_ALL_HOTSPOTS = False  # type: bool
 
+# Toggle faking ldap auth using fakeldap.
+# ZulipLDAPAuthBackend has not been added to AUTHENTICATION_BACKENDS
+# here, user has to add that to enable fake LDAP server.
+FAKE_LDAP_AUTH_ENABLED = True
+
+# Three modes are allowed.
+#   (A) If users' email addresses are in LDAP and used as username.
+#   (B) If LDAP only has usernames but email addresses are of the form
+#       username@example.com
+#   (C) If LDAP username are completely unrelated to email addresses,
+# Refer `prod_settings_template.py` for description of the modes.
+FAKE_LDAP_MODE = 'a'
```

Comment:
I think we could get rid of the `FAKE_LDAP_AUTH_ENABLED` setting, an instead just have a `FAKE_LDAP_MODE` setting that has a default value of `None` (to encode not enabled).